### PR TITLE
[KeyVault] `az keyvault key import`: Support `--curve` parameter for importing BYOK keys

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**

**Testing Guide**

This PR fixes https://github.com/Azure/azure-cli-pr/issues/121

It provides the `--curve` paramter in `az keyvault key import` command.
```
--curve          : The curve name of the key to import (only for BYOK).  Allowed values:
                   P-256, P-256K, P-384, P-521.
```

The command below could be used for key importing after the fix.
```
az keyvault key import --vault-name MyVault --name KeyName --byok-file BYOKFilePath --kty EC --curve P-256
```
The behavior of `--kty` and `--curve` paramters.
- `--kty` has the default value `RSA`
- `--curve` should be specified together with `--kty EC`

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/_params.py [hunk 1]
```diff
@@ -347,8 +353,10 @@ class CLIKeyTypeForBYOKImport(str, Enum):
                    help='Elliptic curve name. For valid values, see: https://docs.microsoft.com/en-us/rest/api/keyvault/createkey/createkey#jsonwebkeycurvename')
 
     with self.argument_context('keyvault key import') as c:
-        c.argument('kty', arg_type=get_enum_type(CLIKeyTypeForBYOKImport),
+        c.argument('kty', arg_type=get_enum_type(CLIKeyTypeForBYOKImport), validator=validate_key_import_type,
                    help='The type of key to import (only for BYOK).')
+        c.argument('curve', arg_type=get_enum_type(CLIJsonWebKeyCurveName), validator=validate_key_import_type,
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/keyvault/_params.py:358]: how about `crv` to be consistent with Json payload?

File: src/azure-cli/azure/cli/command_modules/keyvault/_validators.py [hunk 2]
```diff
@@ -184,6 +184,16 @@ def validate_key_import_source(ns):
...
 
 
+def validate_key_import_type(ns):
+    # Default value of kty is: RSA
+    kty = getattr(ns, 'kty', None)
+    crv = getattr(ns, 'curve', None)
+
+    if (kty == 'EC' and crv is None) or (kty != 'EC' and crv):
+        from azure.cli.core.azclierror import ValidationError
+        raise ValidationError('parameter --curve should be specified when key type is EC.')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/keyvault/_validators.py:194]:
```suggestion
        raise ValidationError('parameter --curve should be specified when key type --kty is EC.')
```

File: src/azure-cli/azure/cli/command_modules/keyvault/_params.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py:358]: Using `curve` here is for the consistentency with existing command `az keyvault key create`, where it uses `--curve` https://github.com/Azure/azure-cli/pull/16593/files#diff-cb5cdc067256b2d2d12c62d95973393cdf48d1105d65833453fab39cefc01fffL346

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/keyvault/_validators.py [hunk 3]
```diff
@@ -184,6 +184,16 @@ def validate_key_import_source(ns):
         raise ValueError('--pem-password must be used with --pem-file or --pem-string')
 
 
+def validate_key_import_type(ns):
+    # Default value of kty is: RSA
+    kty = getattr(ns, 'kty', None)
+    crv = getattr(ns, 'curve', None)
+
+    if (kty == 'EC' and crv is None) or (kty != 'EC' and crv):
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/keyvault/_validators.py:192]: add a `lower()`?

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/keyvault/_validators.py:192]: We don't accept the lower case, it also applies to other existing commands.

**Review by qwordy — APPROVED** [review]

**Pull request merged**